src/coreclr/jit/switchrecognition.cpp

```diff
@@ -252,32 +267,29 @@ bool Compiler::optSwitchDetectAndConvert(BasicBlock* firstBlock, bool testingFor
                 if (testValueIndex == SWITCH_MAX_DISTANCE)
                 {
                     // Too many suitable tests found - stop and process what we already have.
-                    return !testingForConversion &&
-                           optSwitchConvert(firstBlock, testValueIndex, testValues, falseLikelihood, variableNode);
+                    goto optSwitchConvert;
```

Comment:
I don't think this goto is preferred over the existing minor amount of duplication.